.github/workflows/build_deploy.yml

```diff
@@ -0,0 +1,67 @@
+name: Build & Deploy
+
+on:
+  push:
+    branches: [staging]
+
+jobs:
+  build_deploy:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        DISTRIBUTION: ['DISTRIBUTION_0', 'DISTRIBUTION_1']
+
+    env:
+      GH_TOKEN: ${{secrets.GH_TOKEN}}
+      DOCKER_PASSWORD: ${{secrets.DOCKER_PASSWORD}}
+      DOCKER_USERNAME: ${{secrets.DOCKER_USERNAME}}
+      DISTRIBUTION: ${{secrets[matrix.DISTRIBUTION]}}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set BRANCH_NAME environment variable
+        run: echo "::set-env name=BRANCH_NAME::$(git branch --show-current)"
```

Comment:
Why not use `${GITHUB_REF##*/}`?

Reply:
GITHUB_REF is not only the branch, especially for pull requests this expression is wrong.

Reply:
But you are using this on a push, not a pull request.

Reply:
Yep, but for testing I set this to pull request as well. So it's better for debugging and resistant to changes.

Reply:
Also it can be copied to other workflows without changes.